### PR TITLE
[kafka010] Fix: On first run DirectKafkaInputDStream wasn't using backpressure, it was fetching all records instead.

### PR DESCRIPTION
**Pull request by oliviertoupin**
## What changes were proposed in this pull request?

On first run DirectKafkaInputDStream wasn't using backpressure, it was fetching all records instead. Fixed it.

## How was this patch tested?

Test in dev and production our local build of Spark 2.1.0. Backported to master.

Please review http://spark.apache.org/contributing.html before opening a pull request.

### CODE REVIEW

File: external/kafka-0-10/src/main/scala/org/apache/spark/streaming/kafka010/DirectKafkaInputDStream.scala [hunk 1]
```diff
@@ -130,8 +130,15 @@ private[spark] class DirectKafkaInputDStream[K, V](
     offsets: Map[TopicPartition, Long]): Option[Map[TopicPartition, Long]] = {
     val estimatedRateLimit = rateController.map(_.getLatestRate())
 
+    // TODO: Fix in rate controller (need SparkConf or initial rate member value)
+    // Fix: Rate controller first value is -1, if -1 return use the default rate
+    val fixedEstimatedRateLimit = estimatedRateLimit match {
```
Comment (srowen) [inline external/kafka-0-10/src/main/scala/org/apache/spark/streaming/kafka010/DirectKafkaInputDStream.scala]: This looks like a bit of a hack. Can you explain this? what's the right fix if so?
See http://spark.apache.org/contributing.html too

Comment (oliviertoupin) [inline external/kafka-0-10/src/main/scala/org/apache/spark/streaming/kafka010/DirectKafkaInputDStream.scala]: Not so much of a hack, but limit reusability. The best place would be directly in the RateEstimator, but we don't have access to the "spark.streaming.backpressure.initialRate".

**Comment by gaborgsomogyi** [comment]
Looks like a shortcut to me as well.
There is already a PR for backpressure:
https://github.com/apache/spark/pull/19431

Could you explain what exactly would like to achieve here?

**Comment by oliviertoupin** [comment]
My PR predate #19431, but essentially fix the same issue. Since #19431 have more traction and is more thorough, I'll close this one, and comment on the new one if necessary.

**Pull request closed without merging**